azurerm/resource_arm_data_factory_v2.go

```diff
@@ -0,0 +1,343 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataFactoryV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataFactoryV2CreateOrUpdate,
+		Read:   resourceArmDataFactoryV2Read,
+		Update: resourceArmDataFactoryV2CreateOrUpdate,
+		Delete: resourceArmDataFactoryV2Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid name for Data Factory, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules`,
+				),
+			},
+
+			"identity": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								"SystemAssigned",
+							}, true),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"github_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"account_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"collaboration_branch": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"host_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"repository_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"root_folder": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				ConflictsWith: []string{"vsts_configuration"},
+			},
+
+			"vsts_configuration": {
```

Comment:
lastCommitId property is missing in both vsts_configuration and github_configuration and I'm not sure if it's needed in the use case in terraform. Basically this is an optional property used to track changes in which commit has been applied to the data factory. The property is populated if you associate a repo with the factory and go to UI portal and publish some changes